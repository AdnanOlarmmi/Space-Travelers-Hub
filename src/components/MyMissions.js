import React from 'react';
import { useSelector } from 'react-redux';

const MyMissions = () => {
  const missions = useSelector((state) => state.missionReducer.missions);

  const filterMissions = missions.filter((mission) => mission.status === true);
  return (
    <table className="mymissions">
      <thead>
        <th>My Missions</th>
      </thead>
      <tbody>
        {filterMissions.map((mission) => <tr key={mission.id}>{mission.name}</tr>)}
      </tbody>
    </table>
  );
};

export default MyMissions;
