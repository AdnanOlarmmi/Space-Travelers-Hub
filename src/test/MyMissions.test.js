import React from 'react';
import render from './Rockets.test';
import MyMissions from '../components/MyMissions';

describe('MyMissions', () => {
  it('render My Missions Components', () => {
    render(<MyMissions />);
    expect(MyMissions).toMatchSnapshot();
  });
});
