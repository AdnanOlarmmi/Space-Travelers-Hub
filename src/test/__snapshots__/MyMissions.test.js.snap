// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMissions render My Missions Components 1`] = `[Function]`;

exports[`MyRockets render My Rockets Components 1`] = `[Function]`;

exports[`Rockets Check if button works 1`] = `[Function]`;

exports[`Rockets render My Rockets Components 1`] = `[Function]`;
